core/scripts/chaincli/handler/handler.go

```diff
@@ -130,14 +153,225 @@ func (k *Keeper) sendEth(ctx context.Context, to common.Address, amount *big.Int
 	return nil
 }
 
-func (h *baseHandler) waitDeployment(ctx context.Context, tx *types.Transaction) {
+func (h *baseHandler) waitDeployment(ctx context.Context, tx *ethtypes.Transaction) {
 	if _, err := bind.WaitDeployed(ctx, h.client, tx); err != nil {
 		log.Fatal("WaitDeployed failed: ", err)
 	}
 }
 
-func (h *baseHandler) waitTx(ctx context.Context, tx *types.Transaction) {
+func (h *baseHandler) waitTx(ctx context.Context, tx *ethtypes.Transaction) {
 	if _, err := bind.WaitMined(ctx, h.client, tx); err != nil {
 		log.Fatal("WaitDeployed failed: ", err)
 	}
 }
+
+func (h *baseHandler) launchChainlinkNode(ctx context.Context, port int, extraEnvVars ...string) (string, func(), error) {
+	// Create docker client to launch nodes
+	dockerClient, err := client.NewClientWithOpts(client.WithAPIVersionNegotiation())
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to create docker client from env: %s", err)
+	}
+
+	// Make sure everything works well
+	if _, err = dockerClient.Ping(ctx); err != nil {
+		return "", nil, fmt.Errorf("failed to ping docker server: %s", err)
+	}
+
+	// Pull DB image if needed
+	var out io.ReadCloser
+	if _, _, err = dockerClient.ImageInspectWithRaw(ctx, defaultPOSTGRESImage); err != nil {
+		log.Println("Pulling Postgres docker image...")
+		if out, err = dockerClient.ImagePull(ctx, defaultPOSTGRESImage, types.ImagePullOptions{}); err != nil {
+			return "", nil, fmt.Errorf("failed to pull Postgres image: %s", err)
+		}
+		out.Close()
+		log.Println("Postgres docker image successfully pulled!")
+	}
+
+	// Create DB container
+	dbContainerResp, err := dockerClient.ContainerCreate(ctx, &container.Config{
+		Image: defaultPOSTGRESImage,
+		Cmd:   []string{"postgres", "-c", `max_connections=1000`},
+		Env: []string{
+			"POSTGRES_USER=postgres",
+			"POSTGRES_PASSWORD=development_password",
+		},
+		ExposedPorts: nat.PortSet{"5432": struct{}{}},
+	}, nil, &network.NetworkingConfig{}, nil, "")
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to create Postgres container: %s", err)
+	}
+
+	// Start container
+	if err = dockerClient.ContainerStart(ctx, dbContainerResp.ID, types.ContainerStartOptions{}); err != nil {
+		return "", nil, fmt.Errorf("failed to start DB container: %s", err)
+	}
+	log.Println("Postgres docker container successfully created and started: ", dbContainerResp.ID)
+
+	dbContainerInspect, err := dockerClient.ContainerInspect(ctx, dbContainerResp.ID)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to inspect Postgres container: %s", err)
+	}
+
+	time.Sleep(time.Second * 10)
```

Comment:
hacky, but might be fine :p

Reply:
This is not a new code as well. Just moved to another file.